scripts/o3de/tests/test_download.py

```diff
@@ -0,0 +1,263 @@
+                custom_mock.getcode.return_value = 200
+                custom_mock.read.return_value = 0
+                custom_mock.headers = {'content-length': 0}
+                custom_mock.__enter__.return_value = custom_mock
+            else:
+                raise urllib.error.HTTPError(url_str, 404, "Not found", {}, 0)
+
+            return custom_mock
+            
+        def mocked_extract(path=None, members=None, pwd=None):
+            self.extracted_gem_path = path
+            self.extracted_gem_json = gem_data
+            self.created_files.append(path / 'gem.json')
+
+        def mocked_open(path, mode = '', *args, **kwargs):
+            file_data = zip_data.encode('utf-8')
+            if pathlib.Path(path).name == TEST_O3DE_REPO_FILE_NAME:
+                file_data = TEST_O3DE_REPO_WITH_OBJECTS_JSON_PAYLOAD
+            elif pathlib.Path(path).name == TEST_O3DE_REPO_GEM_FILE_NAME or \
+                pathlib.Path(path).name == TEST_O3DE_LOCAL_REPO_GEM_FILE_NAME or \
+                pathlib.Path(path).name == 'gem.json':
+                    file_data = gem_data
+            elif path == pathlib.Path(self.extracted_gem_path) / 'gem.json':
+                file_data = self.extracted_gem_json
+            mockedopen = mock_open(mock=MagicMock(), read_data=file_data)
+            if 'w' in mode:
+                self.created_files.append(path)
+
+            file_obj = mockedopen(self, *args, **kwargs)
+            file_obj.extractall = mocked_extract
+            file_obj.open = mocked_open
+            return file_obj
+
+        def mocked_isfile(path):
+            matches = [pathlib.Path(x).name for x in self.created_files if pathlib.Path(x) == pathlib.Path(path)]
+            if len(matches) != 0:
+                return True
+            else:
+                return False
+
+        def get_cache_folder():
+            return pathlib.Path('C:/Users/testuser/.o3de/cache')
+            
+        def mocked_is_zipfile(path):
+            return True
+
+        def mocked_copy(origin, dest):
+            if mocked_isfile(origin):
+                self.created_files.append(dest)
+
+        download_callback_called = False
+        def download_callback(downloaded, total_size):
+            download_callback_called = True
+
+        with patch('o3de.manifest.load_o3de_manifest', side_effect=load_o3de_manifest) as _1,\
+                patch('o3de.manifest.save_o3de_manifest', side_effect=save_o3de_manifest) as _2, \
+                patch('o3de.manifest.get_o3de_cache_folder', side_effect=get_cache_folder) as _3, \
+                patch('urllib.request.urlopen', side_effect=mocked_requests_get) as _4, \
+                patch('pathlib.Path.open', mocked_open) as _5, \
+                patch('pathlib.Path.is_file', mocked_isfile) as _6, \
+                patch('pathlib.Path.mkdir') as _7, \
+                patch('pathlib.Path.unlink') as _8, \
+                patch('zipfile.is_zipfile', mocked_is_zipfile) as _9, \
```

Comment:
Instead of using a method here you can set the `return_value` keyword value to True
```suggestion
                patch('zipfile.is_zipfile', return_value=True) as _9, \
```

Reply:
Good catch, fixed